R/map_metadata_to_colours.R

```diff
@@ -0,0 +1,154 @@
+#' @title Assign a colour palette to metadata columns automatically and consistently.
+#'
+#' @param metadataColumns Names of the metadata columns to assign colours for.
+#' @param these_samples_metadata Metadata for just the samples you need colours for.
+#' @param column_alias A list of column_names with values indicating what gambl colour scheme they should use (e.g. pos_neg, pathology, lymphgen).
+#' @param as_vector Boolean statement that is set to TRUE per default.
+#' @param verbose Set to TRUE to enable verbose mode (debugging messages.
+#' @param annoAlpha Optional alpha to apply to annotation colours.
+#'
+#' @return Either a vector or list of colours.
+#'
+#' @import dplyr ggsci
+#'
+#' @examples
+#' #get metadata
+#' all_meta = get_gambl_metadata()
```

Comment:
I would suggest updating the example to use the bundled metadata in **GAMBLR.data** (GAMBLR.data::gambl_metadata) instead of calling `get_gambl_metadata` here.

Reply:
Good point! But won't examples like this be fixed once we have a `get_gambl_metadata` function in GAMBLR.data that would call `GAMBLR.results::get_gambl_metadata` if GAMBLR.results is installed, and if not, would call `GAMBLR.data::gambl_metadata` internally? Or something like this?

Maybe we just add the `GAMBLR.data::` prefix in this example once that change is implemented.

Reply:
That's true, 100% I was just worried that GAMBLR.results would be a dependency until the GAMBLR.data PR is in master on that repo.

Reply:
Right. My guess is, by now, GAMLBR.viz has no idea where `get_gambl_metadata` comes from. Let me run `devtools::check` again (I had run it with `args = '--no-examples'`) to see what it says.

Reply:
There is no mention of "get_gambl_metadata` or `GAMBLR.results` in `devtools::check` output. 

Do you think we could keep this as is and fix it later with that GAMBLR.data PR, or make it GAMBLR.data::gambl_metadata until there?

Reply:
I think we can leave it as is for now. I am guessing that there will be many more edits of this kind once everything is configured correctly. I am gonna go ahead and approve this PR so we can make more progress with GAMBLR.results. Thanks Vlad!

Reply:
I think we can leave it as is for now. I am guessing that there will be many more edits of this kind once everything is configured correctly. I am gonna go ahead and approve this PR so we can make more progress with GAMBLR.results. Thanks Vlad!

Reply:
The same that I was thinking. Thanks.